crates/transaction-pool/src/pool/pending.rs

```diff
@@ -223,7 +230,11 @@ impl<T: TransactionOrdering> PendingPool<T> {
         }
         self.all.insert(tx.clone());
 
-        self.by_id.insert(tx_id, tx);
+        self.by_id.insert(tx_id, tx.clone());
+        // send the new transaction to any existing pendingpool snapshot iterators
+        // drop the Result value returned as Err effectively means there are no subscribers
+        // and Ok just returns the # of subscribed recievers
+        let _ = self.new_transaction_notifier.send(tx);
```

Comment:
we assume the most common case is that there are no receivers in which case we don't need to send anything, so lets move this above the `self.by_id.insert` and check for `receiver_count > 0` this prevents 1 clone

Reply:
makes sense, modified